firedrake/formmanipulation.py

```diff
@@ -127,6 +128,59 @@ def argument(self, o):
                          for j in numpy.ndindex(V_is[i].value_shape)]
         return self._arg_cache.setdefault(o, as_vector(args))
 
+    def cofunction(self, o):
```

Comment:
It is just a question. Where is this function called?

Reply:
`ufl.MultiFunction` seems to be essentially a hand-rolled vtable (`Function` as in "multiple functions", not `Function` in the finite element sense), so when `ExtractSubBlocks` calls `map_integrand_dags(self, form)`, the form DAG is traversed and at each node it will look for a method on `ExtractSubBlocks` called `type(node).__name__.lower()`, i.e. `Argument` becomes `argument`, and `Expr` becomes `expr` etc.

So when it gets to a `Cofunction`, it will looks for a method on `ExtractSubBlocks` called `cofunction`. This was missing which was the error that was previously thrown.